code/modules/food_and_drinks/food/customizables.dm

```diff
@@ -118,8 +118,8 @@
 		if(INGREDIENTS_STACKPLUSTOP)
 			filling.pixel_x = rand(-1,1)
 			filling.pixel_y = 2 * ingredients.len - 1
-			if(our_overlays)
-				our_overlays.Cut(ingredients.len)	//???, add overlay calls later in this proc will queue the compile if necessary
+			if(overlays)
+				overlays -= overlays[ingredients.len]
```

Comment:
these are triggering internal overlay rebuilds every time they change the list

Reply:
The only difference is that it does it immediately now, as opposed to a nanosecond later when it gets compiled under the current system. Changing the constituents of "our overlay" still triggers an overlay update on the next SSoverlay fire.

Reply:
the idea was that if there was a lot of overlay churn on an object, the actual update would only happen once a tick rather than several times a tick.

this can be refactor later though

Reply:
also without ssoverlays people coudl crash the server by spamming a welder helmet/other overlay update thing so uh